CMakeLists.txt

```diff
@@ -729,6 +729,11 @@ if(WITH_RADOSGW AND WITH_RADOSGW_LUA_PACKAGES)
   list(APPEND BOOST_COMPONENTS filesystem)
 endif()
 
+if(WITH_LIBCEPHFS)
+  find_package(ICU REQUIRED COMPONENTS uc i18n)
+  list(APPEND BOOST_COMPONENTS locale)
```

Comment:
the `client` target depends on `Boost::locale`, but is not conditional on `WITH_LIBCEPHFS`. i raised https://github.com/ceph/ceph/pull/62108 to fix compilation when `WITH_LIBCEPHFS=OFF`